source/common/network/address_impl.cc

```diff
@@ -260,6 +272,7 @@ Ipv6Instance::Ipv6Instance(const std::string& address, uint32_t port,
   // Just in case address is in a non-canonical format, format from network address.
   ip_.friendly_address_ = ip_.ipv6_.makeFriendlyAddress();
   friendly_name_ = fmt::format("[{}]:{}", ip_.friendly_address_, ip_.port());
+  ip_.ipv6_.v6only_ = v6only;
```

Comment:
Not your code, but is there any reason this part of the function can't reuse initHelper()? (I didn't look carefully just curious, seems like some duplication.)

Reply:
Sure that's fine. No worries.